railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -50,10 +50,10 @@ def gitignore
     end
 
     def lib
-      template "lib/%name%.rb"
-      template "lib/tasks/%name%_tasks.rake"
-      template "lib/%name%/version.rb"
-      template "lib/%name%/engine.rb" if engine?
+      template "lib/%underscored_name%.rb"
```

Comment:
Shouldn't this be `lib/%namespaced_name%.rb`?

If you generate an engine called "foo-bar", you should `require foo/bar`, not `require foo_bar`, no?